### PR TITLE
feat: add GM plugin

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### CODE REVIEW

File: ci/utils/linux-common-runnner.sh [hunk 1]
```diff
@@ -54,11 +54,15 @@ install_module() {
...
+    openresty -V
+
     ./bin/apisix init
     ./bin/apisix init_etcd
 
+    export PATH=/usr/local/tongsuo/bin:$PATH
+    openssl version
     git submodule update --init --recursive
-    FLUSH_ETCD=1 prove -I../test-nginx/lib -I./ -r t/gm
+    FLUSH_ETCD=1 prove -I../test-nginx/lib -I./ -r t/gm || cat t/servroot/logs/error.log
```
Comment (soulbird) [inline ci/utils/linux-common-runnner.sh]: The test fails, but it looks like `||` hides the error.

File: t/gm/gm.t [hunk 2]
```diff
@@ -28,14 +35,122 @@ run_tests;
...
+        ngx.say(body)
     }
+}
+--- response_body
+passed
+
+
+
+=== TEST 2: hit
+--- LAST
```
Comment (soulbird) [inline t/gm/gm.t]: Should remove this.

File: t/gm/gm.t [hunk 3]
```diff
@@ -28,14 +35,122 @@ run_tests;
...
+    }
+}
+--- error_code: 400
+--- response_body
+{"error_msg":"sign cert\/key are required"}
+
+
+
+=== TEST 4: hit with gm disabled
+--- extra_yaml_config
```
Comment (soulbird) [inline t/gm/gm.t:146]: Where is `gm` disabled?

File: ci/utils/linux-common-runnner.sh (hunk 1 above)
Comment (soulbird) [reply ci/utils/linux-common-runnner.sh]: https://github.com/api7/apisix-plugin-gm/pull/3/checks#step:10:64

Comment (spacewander) [reply ci/utils/linux-common-runnner.sh]: Sorry, I forgot to remove debug directive last night

File: t/gm/gm.t (hunk 3 above)
Comment (spacewander) [reply t/gm/gm.t:146]: Here we use an empty `extra_yaml_config` so `gm` plugin is not enabled by `extra_yaml_config`.

**Comment by spacewander** [comment]
@soulbird 
CI passes now

**Review by soulbird — APPROVED** [review]

**Pull request merged**